beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1771,9 +1741,36 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         slot: Slot,
         validator_graffiti: Option<Graffiti>,
     ) -> Result<BeaconBlockAndState<T::EthSpec>, BlockProductionError> {
-        let state = self
-            .state_at_slot(slot - 1, StateSkipConfig::WithStateRoots)
```

Comment:
Are you aware that the `slot - 1` has been removed and that we will no longer be able to produce blocks from slots earlier than the head block?

Reply:
Yeah I did that intentionally, I'll message you

Reply:
Good catch! I've restored the `slot - 1` with a warning, as we discussed.

I think this will be particularly relevant on the first slot of an epoch when there might be two seemingly legitimate proposers because of propagation delay of the last block of the previous epoch.